### PR TITLE
HPCC-14367 Support target alias names for roxies behind a VIP

### PR DESCRIPTION
**Pull request by afishbeck**
Roxies accessed from behind a vip usually use seperate target names to
represent the same set of queries.  Each roxie can also run more than
one target at the same time.  A way is needed to specify that any of the
targets representing the same set of queries can be used.

Allowing targets that represent the same set of queries to have a shared
alias will allow callers to select the right set of queries when calling
through a VIP.

Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

### CODE REVIEW

**Comment by afishbeck** [comment]
Example environment.xml config:
```
  <Topology build="_" buildSet="topology" name="topology">
   <Cluster name="roxie" prefix="roxie" alias="roxie111">
    <EclCCServerProcess process="myeclccserver"/>
    <EclSchedulerProcess process="myeclscheduler"/>
    <RoxieCluster process="myroxie"/>
   </Cluster>
   <Cluster name="roxie2" prefix="roxie2" alias="roxie222">
    <EclCCServerProcess process="myeclccserver"/>
    <EclSchedulerProcess process="myeclscheduler"/>
    <RoxieCluster process="myroxie"/>
   </Cluster>
  </Topology>
```

File: common/workunit/workunit.hpp [hunk 1]
```diff
@@ -527,6 +527,7 @@ interface IConstWUClusterInfo : extends IInterface
     virtual unsigned getRoxieRedundancy() const = 0;
     virtual unsigned getChannelsPerNode() const = 0;
     virtual int getRoxieReplicateOffset() const = 0;
+    virtual IStringVal & getAlias(IStringVal & str) const = 0;
```
Comment (richardkchapman) [inline common/workunit/workunit.hpp]: I want to phase out use of IStringVal... any reason this can't be a const char * queryAlias() method?

**Comment by richardkchapman** [comment]
@wangkx Please review

**Comment by wangkx** [comment]
@afishbeck back to you.

File: common/workunit/workunit.hpp (hunk 1 above)
Comment (afishbeck) [inline common/workunit/workunit.hpp]: I'll change it.  Just my habit of emulating style of surrounding code.

**Comment by afishbeck** [comment]
@richardkchapman updated based on review

**Pull request merged**